lib/logging/addon/components/logging/new-edit/component.js

```diff
@@ -1,82 +1,269 @@
+        setTimeout(() => {
+          set(this, 'tested', false);
+        }, 3000);
       });
+    },
+    cancel() {
+      setProperties(this, {
+        targetType:    get(this, 'preTargetType'),
+        pasteOrUpload: false,
+      })
+    },
 
-      if (targetType === 'none') {
-        this._super(cb);
+    showPaste() {
+      setProperties(this, {
+        preTargetType: get(this, 'targetType'),
+        targetType:    'customTarget',
+        pasteOrUpload: true,
+      })
+    },
+  },
+
+  targetTypeChange: observer('targetType', function() {
+    set(this, 'errors', [])
+  }),
+
+  headerLabel: computed('pageScope', function() {
+    const ps = get(this, 'pageScope');
 
-        return;
+    if (ps === 'cluster') {
+      return 'loggingPage.clusterHeader';
+    } else {
+      return 'loggingPage.projectHeader';
+    }
+  }),
+
+  isClusterLevel: computed('pageScope', function() {
+    return get(this, 'pageScope') === 'cluster';
+  }),
+
+  saveDisabled: computed('originalModel.{id,targetType}', 'targetType', 'pasteOrUpload', function() {
+    return get(this, 'originalModel.targetType') === 'none'
+      && get(this, 'targetType') === 'none'
+      && !get(this, 'pasteOrUpload');
+  }),
+
+  willSave() {
+    let targetType = get(this, 'targetType');
+    const pageScope = this.get('pageScope');
+    const model = get(this, 'model');
+    const intl = get(this, 'intl')
+
+    set(model, 'clusterId', get(this, 'cluster.id'));
+    if (pageScope === 'project') {
+      set(model, 'projectId', get(this, 'project.id'));
+    }
+
+    setProperties(model, {
+      elasticsearchConfig:    null,
+      splunkConfig:           null,
+      kafkaConfig:            null,
+      syslogConfig:           null,
+      fluentForwarderConfig:  null,
+      customTargetConfig:     null,
+    });
+
+    if (targetType === 'none') {
+      return true;
+    }
+
+    set(this, 'errors', [])
+    const errors = get(this, 'errors') || []
+
+    if (get(this, 'pasteOrUpload')) {
+      set(model, 'customTargetConfig', {})
+      set(model, 'customTargetConfig.content', get(this, 'customContent'))
+      let fileObj = {}
+      const customContent = get(this, 'customContent')
+      const myString = customContent.replace(/(\r\n|\n|\r)/gm, '<br />');
+      const keyAndValue = myString.split('<br />').filter((f) => f !== '<br />').filter((f = '') => !f.startsWith('#') && !f.startsWith('<'))
+
+      keyAndValue.map((item = '') => {
+        const arr = item.split(' ').filter((f) => f !== '')
+
+        console.log(arr, 'arr')
+        set(fileObj, arr[0], arr[1])
+      })
+
+      const targetType = fileObj['@type']
+      const types = ['elasticsearch', 'splunk_hec', 'remote_syslog', 'kafka_buffered', 'forward']
+
+      if (!types.includes(targetType)) {
+        errors.pushObject(intl.t('loggingPage.customTarget.type.error'))
       }
 
-      set(this, 'errors', [])
-      const errors = get(this, 'errors') || []
+      switch (targetType) {
```

Comment:
All of the blocks in this switch use the same object for the `xxConfig`, it would be cleaner to create the config object once before the switch opens and use that in the correct switch block.